castervoice/lib/ccr/core/nav.py

```diff
@@ -95,7 +96,10 @@ class NavigationNon(MappingRule):
         "undo [<n>]":
             R(Key("c-z"), rdescript="Core: Undo")*Repeat(extra="n"),
         "redo [<n>]":
-            R(Key("c-y"), rdescript="Core: Redo")*Repeat(extra="n"),
+            R(ContextAction(default=Key("c-y")*Repeat(extra="n"), actions=[
+                # Use cs-z for rstudio
+                (AppContext(executable="rstudio"), Key("cs-z")*Repeat(extra="n")),
```

Comment:
This should be in app grammar. Jetbrains is also cs-z for example

Reply:
Using contextaction is an alternative to using app grammars.

Reply:
Will add jetbrains. Does c-y not work there?

Reply:
I meant from a design perspective since otherwise the code will get polluted with app specific if statements

Reply:
In this case where there are just a few exceptions to the common key mapping, I think I prefer ContextAction to adding the exceptions to App grammars. Let's see how it goes doing it this way and if the core grammars start to engorge we can move stuff into app files.

Reply:
@lexxish, The apps can be specified as a list as well, so adding a new app to a ContextAction is super compact. Now that I know this, I'm skeptical that it's better to have app grammars at all.